src/item_factory.cpp

```diff
@@ -1857,15 +1855,28 @@ void Item_factory::load( islot_comestible &slot, const JsonObject &jo, const std
             specific_heat_solid += material_id( m )->specific_heat_solid();
             specific_heat_liquid += material_id( m )->specific_heat_liquid();
             latent_heat += material_id( m )->latent_heat();
-            is_junkfood = is_junkfood || m == "junk";
         }
         // Average based on number of materials.
         slot.specific_heat_liquid = specific_heat_liquid / jo.get_tags( "material" ).size();
         slot.specific_heat_solid = specific_heat_solid / jo.get_tags( "material" ).size();
         slot.latent_heat = latent_heat / jo.get_tags( "material" ).size();
     }
 
-    if( is_junkfood ) { // Junk food never gets old by default, but this can still be overriden.
+    bool is_not_boring = false;
+    if (jo.has_member("primary_material")) {
+        std::string mat = jo.get_string("primary_material");
+        is_not_boring = is_not_boring || mat == "junk";
+    }
+    if (jo.has_member("material")) {
+        for (auto& m : jo.get_tags("material")) {
```

Comment:
Please replace auto with the proper type.

Reply:
What's the proper type? String? I didn't write that, just copied it from the existing material check.

Reply:
Looks like std::string.

Reply:
It didn't crash on item load, so that seems to be it. Thanks. Switched both of them.

Reply:
It would fail to compile if you put the wrong type, so you can know you've got it right.